### PR TITLE
MAT-6198 Exported transaction test case bundles are invalid due to MeasureReport missing request component

### PR DESCRIPTION
**Pull request by adongare**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-6198](https://jira.cms.gov/browse/MAT-6198)
(Optional) Related Tickets:
[MAT-6237](https://jira.cms.gov/browse/MAT-6237) QI-Core Measure Export Bundle Updates
### Summary
- Measure bundles are transaction bundles- should have a request component for all the resources in the bundle always. HTTP verb is POST
- Test case bundles can be transaction or non-transactional. for transactional test case bundles MeasureReport should have a request component. HTTP verb is POST.
- all resources would have a full URL that starts with `https://madie.cms.gov/`
- frontend PR: https://github.com/MeasureAuthoringTool/madie-patient/pull/405

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java [hunk 1]
```diff
@@ -90,31 +90,20 @@ public Bundle createMeasureBundle(
   public List<Bundle.BundleEntryComponent> createBundleComponentsForLibrariesOfMadieMeasure(
       Measure madieMeasure, final String bundleType, final String accessToken) {
     Library library = getMeasureLibraryResourceForMadieMeasure(madieMeasure);
-    Bundle.BundleEntryComponent mainLibraryBundleComponent = getBundleEntryComponent(library);
+    Bundle.BundleEntryComponent mainLibraryBundleComponent =
+        FhirResourceHelpers.getBundleEntryComponent(library, "Transaction");
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:94]: shouldn't the type come from what is specified in the bundle? If the bundle is a collection type, we wouldn't want to add the "request" section to the entry

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java [hunk 2]
```diff
@@ -90,31 +90,20 @@ public Bundle createMeasureBundle(
...
     Map<String, Library> includedLibraryMap = new HashMap<>();
     libraryService.getIncludedLibraries(
         madieMeasure.getCql(), includedLibraryMap, bundleType, accessToken);
     List<Bundle.BundleEntryComponent> libraryBundleComponents =
         includedLibraryMap.values().stream()
-            .map(this::getBundleEntryComponent)
+            .map((lib) -> FhirResourceHelpers.getBundleEntryComponent(lib, "Transaction"))
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:100]: same comment as above regarding bundle type and request entry

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:94]: this is for measure bundle. it is always transactional, at least for now.

File: src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java (hunk 2 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/MeasureBundleService.java:100]: this is for the measure bundle as well.

**Review by nmorasb — APPROVED** [review]

**Pull request merged**